contracts/partners/giveth/GivethMinter.sol

```diff
@@ -0,0 +1,208 @@
+      require(sentPgfFee, "Failed to send pgfFee");
+
+      selectedPrice -= pgfFeePayment;
+    }
+
+    // send developer fee
+    if (devAddress != address(0)) {
+      uint256 devPayment = selectedPrice / 2;
+      (bool sentDevFee, ) = payable(devAddress).call{value: devPayment}("");
+      require(sentDevFee, "Failed to send developer fee");
+    }
+
+    // send the rest to TLD owner
+    (bool sent, ) = payable(tldContract.owner()).call{value: address(this).balance}("");
+    require(sent, "Failed to send domain payment to TLD owner");
+
+    // mint a domain
+    tokenId = tldContract.mint{value: 0}(_domainName, _domainHolder, address(0));
+  }
+
+  // OWNER
+
+  /// @notice This changes the pgfFee in the minter contract
+  function changePgfFee(uint256 _pgfFee) external onlyOwner {
+    require(_pgfFee <= 8000, "Cannot exceed 80%");
+    pgfFee = _pgfFee;
+  }
+
+  /// @notice This changes price in the minter contract
+  function changePrice(uint256 _price, uint256 _chars) external onlyOwner {
+    require(_price > 0, "Cannot be zero");
+
+    if (_chars == 1) {
+      price1char = _price;
+    } else if (_chars == 2) {
+      price2char = _price;
+    } else if (_chars == 3) {
+      price3char = _price;
+    } else if (_chars == 4) {
+      price4char = _price;
+    } else if (_chars == 5) {
+      price5char = _price;
+    }
+  }
+
+  /// @notice This changes referral fee in the minter contract
+  function changeReferralFee(uint256 _referralFee) external onlyOwner {
+    require(_referralFee <= 2000, "Cannot exceed 20%");
+    referralFee = _referralFee;
+  }
+
+  function ownerFreeMint(
+    string memory _domainName,
+    address _domainHolder
+  ) external nonReentrant onlyOwner returns(uint256 tokenId) {
+    // mint a domain
+    tokenId = tldContract.mint{value: 0}(_domainName, _domainHolder, address(0));
+  }
+
+  /// @notice Recover any ERC-20 token mistakenly sent to this contract address
+  function recoverERC20(address tokenAddress_, uint256 tokenAmount_, address recipient_) external onlyOwner {
+    IERC20(tokenAddress_).transfer(recipient_, tokenAmount_);
+  }
+
+  /// @notice Recover any ERC-721 token mistakenly sent to this contract address
+  function recoverERC721(address tokenAddress_, uint256 tokenId_, address recipient_) external onlyOwner {
+    IERC721(tokenAddress_).transferFrom(address(this), recipient_, tokenId_);
+  }
+
+  function togglePaused() external onlyOwner {
+    paused = !paused;
+  }
+
+  // withdraw ETH from contract
+  function withdraw() external onlyOwner {
+    (bool success, ) = owner().call{value: address(this).balance}("");
+    require(success, "Failed to withdraw ETH from contract");
+  }
+
+  // OTHER WRITE METHODS
+
+  /// @notice This changes the pool address in the minter contract
+  function changePgfAddress(address _pgfAddress) external {
+    require(_msgSender() == pgfAddress, "Sender is not the pool address owner");
```

Comment:
@chnejohnson It just crossed my mind that the PGF address is the Giveth Matching Pool, which is probably a smart contract, so it cannot change its own address here. 

So let's move this function to the Owner Functions section and make it so that only Minter contract owner can change the PGF address 🙂

Reply:
Okay